de-lib/src/main/java/org/iplantc/de/admin/desktop/client/communities/service/impl/AdminCommunityServiceFacadeImpl.java

```diff
@@ -0,0 +1,139 @@
+import static org.iplantc.de.shared.services.BaseServiceCallWrapper.Type.PATCH;
+import static org.iplantc.de.shared.services.BaseServiceCallWrapper.Type.POST;
+
+import org.iplantc.de.admin.desktop.client.communities.service.AdminCommunityServiceFacade;
+import org.iplantc.de.client.models.HasId;
+import org.iplantc.de.client.models.apps.App;
+import org.iplantc.de.client.models.apps.proxy.AppListLoadResult;
+import org.iplantc.de.client.models.apps.proxy.AppSearchAutoBeanFactory;
+import org.iplantc.de.client.models.collaborators.CollaboratorAutoBeanFactory;
+import org.iplantc.de.client.models.collaborators.Subject;
+import org.iplantc.de.client.models.groups.Group;
+import org.iplantc.de.client.models.groups.GroupAutoBeanFactory;
+import org.iplantc.de.client.models.groups.UpdateMemberRequest;
+import org.iplantc.de.client.models.groups.UpdateMemberResult;
+import org.iplantc.de.client.services.converters.DECallbackConverter;
+import org.iplantc.de.client.services.converters.GroupCallbackConverter;
+import org.iplantc.de.client.services.converters.GroupListCallbackConverter;
+import org.iplantc.de.client.services.converters.SubjectMemberListCallbackConverter;
+import org.iplantc.de.client.services.converters.UpdateMemberResultsCallbackConverter;
+import org.iplantc.de.shared.DECallback;
+import org.iplantc.de.shared.services.DiscEnvApiService;
+import org.iplantc.de.shared.services.ServiceCallWrapper;
+
+import com.google.gwt.http.client.URL;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.inject.Inject;
+import com.google.web.bindery.autobean.shared.AutoBeanCodex;
+import com.google.web.bindery.autobean.shared.AutoBeanUtils;
+import com.google.web.bindery.autobean.shared.Splittable;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @author aramsey
+ */
+public class AdminCommunityServiceFacadeImpl implements AdminCommunityServiceFacade {
+
+    private final String ADMIN_COMMUNITIES = "org.iplantc.services.admin.communities";
+    private final String ADMIN_APPS_COMMUNITIES = "org.iplantc.services.admin.apps.communities";
+    @Inject GroupAutoBeanFactory factory;
+    @Inject AppSearchAutoBeanFactory appAutoBeanFactory;
+    @Inject CollaboratorAutoBeanFactory collaboratorAutoBeanFactory;
+    @Inject private DiscEnvApiService deService;
+
+    @Inject
+    AdminCommunityServiceFacadeImpl() {
+    }
+
+    @Override
+    public void getCommunities(AsyncCallback<List<Group>> callback) {
+        String address = ADMIN_COMMUNITIES;
+
+        ServiceCallWrapper wrapper = new ServiceCallWrapper(GET, address);
+        deService.getServiceData(wrapper, new GroupListCallbackConverter(factory, callback));
+    }
+
+    @Override
+    public void getCommunityApps(Group community, DECallback<List<App>> appListCallback) {
+        String address = ADMIN_APPS_COMMUNITIES + "/" + URL.encodePathSegment(community.getDisplayName()) + "/apps";
```

Comment:
So currently this `terrain` endpoint is just a passthrough to the corresponding `apps` endpoint. The `apps` service doesn't know anything about building the full `display_name` for the environment it's running in (`qa` vs. `prod` for example), but `terrain` does, which is why the other `/communities` endpoints only use the short community `name`.

I wonder if it would be better to make this endpoint (and the non- `/admin` version) a facade in `terrain` that only needs the short community `name`. Then this `terrain` endpoint could build the full `display_name` itself for use in the `apps` endpoint 🤔

Reply:
🤔 It would make the service calls at least in the UI more consistent and if the logic is already there, I think it makes sense to change it.  If it ends up complicating terrain though, it's easy enough for the UI to send the `display_name` or short name.